dotCMS/src/main/java/com/dotcms/contenttype/business/uniquefields/extratable/UniqueFieldDataBaseUtil.java

```diff
@@ -500,6 +504,63 @@ public void addPrimaryKeyConstraintsBack() throws DotDataException {
         new DotConnect().setSQL(sqlQuery).loadObjectResults();
     }
 
+    /**
+     * Adds the necessary Indexes to the {@code unique_fields} table in order to improve its
+     * performance as much as possible.
+     *
+     * @throws DotDataException An error occurred when interacting with the database.
+     */
+    public void addTableIndexes() throws DotDataException {
```

Comment:
this seems to be that it is taking a new connection and do not closing it
I would annotated it with ExternalTransaction instead